src/index.ts

```diff
@@ -75,16 +91,22 @@ try {
 
   // Create the index if it doesn't already exist
   const indexList = await pinecone.listIndexes();
-  if (indexList.indexOf({ name: indexName }) === -1) {
-    await pinecone.createIndex({ name: indexName, dimension: 384, waitUntilReady: true })
+  if (!indexList.indexes?.some((index) => index.name === indexName)) {
```

Comment:
Update to handle `listIndexes()` response and the `createIndex()` call below now includes `spec`.